check_changed.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+set -e
+# This script is called by Travis during the install step.
+# It returns 1 if no files where changed. In that case
+# no further building/test is required for this image
+
+IMAGE_NAME=$1
+
+
+echo "Travis event type: $TRAVIS_EVENT_TYPE"
+
+if [ "$TRAVIS_EVENT_TYPE" == "api" ]
+then
+    # trigger via travis dashboard
+    # test all
+    echo "Trigger all tests"
+    exit 0
+fi
+
+
+git remote set-branches --add origin $TRAVIS_BRANCH
+git fetch
+
+cd "$IMAGE_NAME-docker"
+
+CHANGED_FILES=$(git diff --name-status $TRAVIS_BRANCH~1...$TRAVIS_BRANCH .)
+if [ -z "$CHANGED_FILES" ]
+then
+    # nothing changed, skip building
+    exit 1 
+fi
+echo "There were changes in $IMAGE_NAME changes: $CHANGED_FILES"
```

Comment:
It was a bit messy because the check_changes script used a return code to indicate if something changed or not. If a real error occurred in the script (eg. git timeout) this was mistakenly interpreted as "no changes".

I moved 'travis_terminate'  into the script. No more mishandling of bash return codes.